### PR TITLE
Remove SeccompLevel from Jailer

### PR DESCRIPTION
**Pull request by xibz**
Firecracker has since removed specifying the seccomp level in the jailer
and now is specified in Firecracker instead. This change removes the
seccomp level from the jailer and adds it to machine instead.

Signed-off-by: xibz <impactbchang@gmail.com>

TODO:

- [x] Update CI system to build against master as well

#175 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by xibz** [comment]
managed to fix the seccomp issue and add the updated CI test to verify. However, Firecracker is panicking for some odd reason. Here is the issue, https://github.com/firecracker-microvm/firecracker/issues/1557

**Comment by xibz** [comment]
root tests will fail due to Firecracker moving where the api socket location is.

**Comment by IRCody** [comment]
> root tests will fail due to Firecracker moving where the api socket location is.

Is there a reason we can't look in the new location?

**Comment by xibz** [comment]
@IRCody -We are using the new location. Firecracker has changed the default location, which is a breaking change. Which is why it succeeds against master but fails the root tests, since it is looking for the new location

**Comment by IRCody** [comment]
So those tests will pass if we update the firecracker in CI?

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -101,6 +101,9 @@ steps:
       queue: "${BUILDKITE_AGENT_META_DATA_QUEUE:-default}"
       distro: "${BUILDKITE_AGENT_META_DATA_DISTRO}"
       hostname: "${BUILDKITE_AGENT_META_DATA_HOSTNAME}"
+    # TODO: Remove this once v0.21.0 has been released
```
Comment (kzys) [inline .buildkite/pipeline.yml]: Can we have build flags or something to remove `soft_fail`? We are not supporting multiple Firecracker versions, but keeping soft_fail will mask new incompatibilities.

Alternatively we could skip some tests.

Comment (xibz) [reply .buildkite/pipeline.yml]: > keeping soft_fail will mask new incompatibilities

Build flags would also do this?

**Comment by xibz** [comment]
> So those tests will pass if we update the firecracker in CI?

Once Firecracker releases v0.21.0, then yes. However, they haven't done so, but customers are using HEAD of Firecracker and we shouldn't limit ourselves to the most released version, due to this.

**Comment by kzys** [comment]
Here is what we used to have. Probably not really intentional, but due to the fact we didn't have Firecracker master on our build host.

|            | Firecraker v0.X.0 | Firecracker master |
|------------|-------------------|--------------------|
| SDK v0.X.0 | Must Work         | We Don't Know      |
| SDK master | Must Work         | We Don't know      |

And here is what we are going to have.

|            | Firecraker v0.X.0 | Firecracker master |
|------------|-------------------|--------------------|
| SDK v0.X.0 | Must Work         | We Don't Know      |
| SDK master | May Work          | Must Work          |

I agree that making master compatible with Firecracker's latest release is nice to have, but may require some efforts and we don't have customers who specifically ask about.

So I think merging this change is fine. We can always create a branch and cherry-pick if we want to have some fixes for Firecracker 0.20.0 customers.

**Pull request merged**